CEGameApp.API/Controllers/AuthController.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using CEGameApp.API.Data;
+using CEGameApp.API.Dtos;
+using CEGameApp.API.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+
+namespace CEGameApp.API.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class AuthController : ControllerBase
+    {
+        private readonly IAuthRepository _repo;
+        private readonly IConfiguration _config;
+        private readonly IUserRepository _userRepo;
+        private readonly ILoginService _loginService;
+
+        public AuthController(IAuthRepository repo, IConfiguration config, IUserRepository userRepo, ILoginService loginService)
+        {
+            _repo = repo;
+            _config = config;
+            _userRepo = userRepo;
+            _loginService = loginService;
+        }
+
+        [HttpPost("register")]
+        public async Task<IActionResult> Register(UserForRegisterDto userForRegisterDto)
+        {            
+            if (await _repo.UserExists(userForRegisterDto.UserName.ToLower()))
+                return BadRequest("Username already exists");
+            await _userRepo.RegisterUser(userForRegisterDto.UserName.ToLower(), userForRegisterDto.Password);
+            
+            return StatusCode(201);  //will change to createdatroute once we have a place to get data from
+        }
+
+
+        [HttpPost("login")]
+        public async Task<IActionResult> Login(UserForLoginDto userForLoginDto)
+        {
+            var userFromRepo = await _repo.Login(userForLoginDto.UserName.ToLower(), userForLoginDto.Password);
```

Comment:
still need to validate the model that is coming in.  Model is a Property of this classes base class.  Model has a property on it called something like IsValid.  that is how you check if your model is valid.  you have validation attributes on your properties but you are not doing anything with it

Reply:
Validation added to UserForLoginDto.